src/lib/translators/mongo.ts

```diff
@@ -545,11 +548,22 @@ const mapper: Partial<StepMatcher<MongoStep>> = {
     },
   }),
   filter: filterstepToMatchstep,
-  formula: (step: Readonly<S.FormulaStep>) => ({
-    $addFields: {
-      [step.new_column]: buildMongoFormulaTree(math.parse(step.formula)),
-    },
-  }),
+  formula: (step: Readonly<S.FormulaStep>) => {
+    // replace all `[column name]` values by a pseudo and store it in `pseudos`
+    // `buildMongoFormulaTree` will re-put the name back
+    const pseudos: { [key: string]: string } = {};
+    const formulaPseudotised = step.formula.replace(/\[.*?\]/g, columName => {
```

Comment:
I think you could do something like:

```suggestion
    const formulaPseudotised = step.formula.replace(/\[(.*?)\]/g, (_match, columName) => {
```

and avoid calling`replace` inside the callback, cf. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_function_as_a_parameter